app/actions/remote/channel.ts

```diff
@@ -49,6 +51,96 @@ export type MyChannelsRequest = {
     error?: unknown;
 }
 
+export type ChannelMembersRequest = {
+    members?: ChannelMembership[];
+    error?: unknown;
+}
+
+export async function removeMemberFromChannel(serverUrl: string, channelId: string, userId: string) {
+    try {
+        const {operator} = DatabaseManager.getServerDatabaseAndOperator(serverUrl);
+        const client = NetworkManager.getClient(serverUrl);
+
+        await client.removeFromChannel(userId, channelId);
+        await deleteChannelMembership(operator, userId, channelId);
+
+        return {error: undefined};
+    } catch (error) {
+        forceLogoutIfNecessary(serverUrl, error as ClientErrorProps);
+        return {error};
+    }
+}
+
+export async function fetchChannelMembersByIds(serverUrl: string, channelId: string, userIds: string[], fetchOnly = false): Promise<ChannelMembersRequest> {
+    try {
+        const client = NetworkManager.getClient(serverUrl);
+        const members = await client.getChannelMembersByIds(channelId, userIds);
+
+        if (!fetchOnly) {
+            const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
+            if (operator && members.length) {
+                const modelPromises: Array<Promise<Model[]>> = [];
+                const membership = members.map((u) => ({
+                    channel_id: channelId,
+                    user_id: u.user_id,
+                    scheme_admin: u.scheme_admin,
+                }));
+                modelPromises.push(operator.handleChannelMembership({
+                    channelMemberships: membership,
+                    prepareRecordsOnly: true,
+                }));
+
+                const models = await Promise.all(modelPromises);
+                await operator.batchRecords(models.flat());
+            }
+        }
+
+        return {members};
+    } catch (error) {
+        forceLogoutIfNecessary(serverUrl, error as ClientError);
+        return {error};
+    }
+}
+export async function updateChannelMemberSchemeRoles(serverUrl: string, channelId: string, userId: string, isSchemeUser: boolean, isSchemeAdmin: boolean) {
+    try {
+        const client = NetworkManager.getClient(serverUrl);
+
+        await client.updateChannelMemberSchemeRoles(channelId, userId, isSchemeUser, isSchemeAdmin);
+        await client.getMemberInChannel(channelId, userId);
+
+        return {error: undefined};
+    } catch (error) {
+        forceLogoutIfNecessary(serverUrl, error as ClientErrorProps);
+        return {error};
+    }
+}
+
+export async function getMemberInChannel(serverUrl: string, channelId: string, userId: string) {
+    try {
+        const client = NetworkManager.getClient(serverUrl);
+        const member = await client.getMemberInChannel(channelId, userId);
+
+        // TODO: confirm with Elias if this is right way to do this since this
+        // is happening on webapp
+        updateLocalUser(serverUrl, member, userId);
```

Comment:
Would be great to have this TODO removed before merging. My two cents here is that this should have a `fetchOnly` and update the user membership based on that.

Reply:
fetchonly Added as an option.